File: go/benchmark.go
package main

import(
	"fmt"
	"os"
	"helper"
//	"strconv"
	"quicksort"
)

func main() {
//	if(os.Args[0] == nil){
//		fmt.Println("File usage: <stringlen> <arraylen>")
//	}

	t1 := os.Args[0]
	t2 := os.Args[1]

	stringlen := 10 //strconv.ParseInt(t1, 0, 64)
	arraylen := 10 //strconv.ParseInt(t2, 0, 64)

	helper.ResetTimer()
	var arr []string = helper.MakeArray(stringlen, arraylen)
	var time  = helper.ReadTimerMilliseconds()
	fmt.Printf("Generated %d strings of length %d in %.4f ms\n", arraylen, stringlen, time)
    	for _, x := range(arr) {
       	fmt.Println(x)
    	}

    	helper.ResetTimer()
    	quicksort.sort(arr, 0, arraylen-1)
    	time  = helper.ReadTimerMilliseconds()
	fmt.Printf("Sorted %d strings of length %d in %.4f ms\n", arraylen, stringlen, time)
    	for _, x := range(arr) {
       	fmt.Println(x)
    	}

}

File: a1_Marshall/java/csci2020u/a1/Benchmark.java
// Un-comment the lines below to verify that the sorting method works properly
package csci2020u.a1;

import csci2020u.a1.Helper;
import java.util.Arrays;

public class Benchmark {
	public static void main(String[] args){
		if(args.length == 0){
			System.out.println("Class usage: java csci2020u.a1.Benchmark <string length> <array length>");
			return;
		}

		int stringLen = Integer.parseInt(args[0]);
		int arrayLen = Integer.parseInt(args[1]);

		Helper.resetTimer();
		String[] arr = Helper.makeArray(stringLen, arrayLen);
//		System.out.println(Arrays.deepToString(arr));
		System.out.printf("Created %d strings of length %d in %.0f milliseconds\n", 
       		arrayLen, stringLen, Helper.readTimerMilliseconds() / 1.0f);

		QuickSorter qs = new QuickSorter();
		Helper.resetTimer();
		qs.QuickSorter(arr);

//		System.out.println(Arrays.deepToString(arr));
		System.out.printf("~Sorted %d strings of length %d in %.0f milliseconds\n\n", 
       		arrayLen, stringLen, Helper.readTimerMilliseconds() / 1.0f);
	}
}